pcsx2/ImGui/FullscreenUI.cpp

```diff
@@ -8663,6 +8678,376 @@ void FullscreenUI::DrawAboutWindow()
+				}
+				else
+				{
+					ImGui::TextUnformatted(FSUI_CSTR("Downloading covers..."));
+				}
+
+				// Show progress bar if we have progress info
+				if (s_cover_downloader_progress_max > 0)
+				{
+					const float progress = static_cast<float>(s_cover_downloader_progress_value) / static_cast<float>(s_cover_downloader_progress_max);
+					const int percent = static_cast<int>(std::round(progress * 100.0f));
+					ImGui::ProgressBar(progress, ImVec2(-1.0f, LayoutScale(30.0f)), fmt::format("{}%", percent).c_str());
+				}
+				else
+				{
+					// Indeterminate progress bar
+					const float fraction = std::fmod(ImGui::GetTime(), 2.0f) * 0.5f;
+					ImGui::ProgressBar(fraction, ImVec2(-1.0f, LayoutScale(30.0f)));
+				}
+
+				ImGui::SetCursorPosY(ImGui::GetCursorPosY() + LayoutScale(8.0f));
+			}
+		}
+
+		ImGui::SetCursorPosY(ImGui::GetCursorPosY() + LayoutScale(20.0f));
+
+		const bool has_urls = (s_cover_downloader_urls_buffer[0] != '\0');
+		bool start_enabled;
+		{
+			std::lock_guard<std::mutex> lock(s_cover_downloader_mutex);
+			start_enabled = !s_cover_downloader_downloading && has_urls;
+		}
+
+		{
+			std::unique_lock<std::mutex> lock(s_cover_downloader_mutex);
+			const bool is_downloading = s_cover_downloader_downloading;
+			lock.unlock();
+
+			if (is_downloading)
+			{
+				if (ActiveButton(FSUI_ICONSTR(ICON_FA_STOP, "Stop"), false))
+				{
+					{
+						std::lock_guard<std::mutex> lock2(s_cover_downloader_mutex);
+						s_cover_downloader_downloading = false;
+					}
+					if (s_cover_downloader_thread && s_cover_downloader_thread->joinable())
+						s_cover_downloader_thread->join();
+					s_cover_downloader_thread.reset();
+				}
+			}
+			else
+			{
+				if (ActiveButton(FSUI_ICONSTR(ICON_FA_PLAY, "Start"), false, start_enabled))
+				{
+					// Check if URLs are valid before starting
+					std::vector<std::string> urls;
+					std::string urls_text(s_cover_downloader_urls_buffer.data());
+					for (const auto& line : StringUtil::SplitString(urls_text, '\n', true))
+					{
+						if (!line.empty())
+							urls.push_back(std::string(line));
+					}
+
+					// Check if all URLs are valid
+					bool has_valid_template = false;
+					for (const auto& url : urls)
+					{
+						if (url.find("${title}") != std::string::npos ||
+							url.find("${filetitle}") != std::string::npos ||
+							url.find("${serial}") != std::string::npos)
+						{
+							has_valid_template = true;
+							break;
+						}
+					}
+
+					if (!has_valid_template)
+					{
+						ShowToast(FSUI_STR("Error"), FSUI_STR("URL template must contain at least one of ${title}, ${filetitle}, or ${serial}."), 5.0f);
+					}
+					else
+					{
+						// Start download in a background thread
+						bool should_start_download = false;
+						{
+							std::lock_guard<std::mutex> lock(s_cover_downloader_mutex);
+							if (!s_cover_downloader_downloading)
+							{
+								s_cover_downloader_downloading = true;
```

Comment:
Just because the worker thread was cancelled, it doesn't mean that the thread knows it's cancelled yet, so this looks like it might be a race. I would wait until the thread is cancelled and joined, and only then reset these variables.